### ISSUE TITLE
diagnostic_table() should be a standalone function

### ISSUE THREAD

**Issue by MattHJensen** [open-source-economics/Tax-Calculator#213]
[diganostic_table() is currently a Calculator method](https://github.com/OpenSourcePolicyCenter/Tax-Calculator/blob/master/taxcalc/calculate.py#L277-L360). Instead, it should be broken into two standalone functions in utils.py. 

1. diagnostic_table() should be a non-mutating function that accepts a calculator and produces a table for a single year, similar to the other table generators in utils.py
2. There should also be a "multi year diagnostics table generator" that accepts a calculator and a number of years, creates a hardcopy of that calculator, and then does something like

```  
df = []
for i in num_years:
    calc.calc_all()
    df1 = diagnostic_table(calc)
    df = df.append(df1)
    calc.iterate_year()
return df
```

**Comment by talumbau** [comment]
Agree. @MattHJensen, is this a task that you see as falling in my bucket or someone else's?

**Comment by MattHJensen** [comment]
I'm on this one.

**Comment by martinholmer** [comment]
Issue #213 has been resolved by pull requests #848 and #852.
